feeds/product.php

```diff
@@ -38,6 +38,151 @@
 
 DfTools::validateSecurityToken(Tools::getValue('dfsec_hash'));
 
+/**
+ * Global storage for all active measurements.
+ *   - $__metrics_data[$label] holds ['time' => float, 'memory' => int, 'peak' => int]
+ *   - $__metrics_stack is an indexed array of labels, used to pop the most recent.
+ */
+$__metrics_data  = [];
+$__metrics_stack = [];
+
+/**
+ * Begin measuring time & memory.
+ *
+ * @param string|null $label  Optional: a custom label for this measurement.
+ *                            If you pass null (or omit), a unique label is generated.
+ * @return string             The label that was used.  (Use the same label in stop_metrics() if you want.)
+ */
+function start_metrics(string $label = null): string
```

Comment:
In order to guarantee compatibility with all the supported PHP version try to avoid type declarations on parameters and returns:

```suggestion
function startMetrics($label = null)
```

With the ones defined in the PHPDocs should be enough :smile: 

Another comment to add is that PrestaShop standards suggest to use camelCase, not snake_case :raised_hands:

Reply:
Oh, this is just for this draft. It was not the idea to publish this part. Sorry for the inconvenience and the time taken to review that part :/